### PR TITLE
Domains Table: Make the domain table row clickable

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/dotcom-forge/issues/6999

## Proposed Changes

* Make the domain table row clickable

## Testing Instructions

* Go to /domains/manage
* Click the domain row
* Make sure it opens the domain management page

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/domains-table/src/domains-table/style.scss [hunk 1]
```diff
@@ -21,7 +21,14 @@
 		--row-gap: 20px;
 
 		display: grid;
-		gap: var(--row-gap) 16px;
```
Comment (arthur791004) [inline packages/domains-table/src/domains-table/style.scss:24]: The gap area is not clickable, so change to padding instead

**Comment by DustyReagan** [comment]
It's working well in general, but I noticed that you can not select a domain with the checkbox without it redirecting.

Screenshot 2024-05-08 at 10 12 46 AM

**Review by lupus2k — APPROVED** [review]
LGTM!

**Review by roo2 — CHANGES_REQUESTED** [review]
The mobile view still doesn't seem to be clickable on the calypso.live site.

https://github.com/Automattic/wp-calypso/assets/22446385/5c48d182-67a5-4e43-a703-d06fbc5df310

**Comment by taipeicoder** [comment]
Seeing the same issues as mentioned by @DustyReagan and @roo2!

**Comment by arthur791004** [comment]
> It's working well in general, but I noticed that you can not select a domain with the checkbox without it redirecting.

Addressed by https://github.com/Automattic/wp-calypso/pull/90441/commits/36a52b1c6ea1aea1925b74dbb7214805fc6d5e2f

> The mobile view still doesn't seem to be clickable on the calypso.live site.

Addressed by https://github.com/Automattic/wp-calypso/pull/90441/commits/cbef28a58bae01aacc47bf4449f22f24590dac39. I didn't notice we use the different component on mobile 😓

**Comment by taipeicoder** [comment]
@arthur791004 I'm not able to see any checkboxes in mobile resolution, is this expected?
Screenshot 2024-05-09 at 4 32 59 PM

**Comment by arthur791004** [comment]
> I'm not able to see any checkboxes in mobile resolution, is this expected?

It's the current behavior when you land on the page on mobile. You have to click "..." to enable the bulk actions.

image

**Review by allilevine — APPROVED** [review]
This is testing well for me! ✅ 

I noticed the checkbox and text are closer than in the current styles, but they're not overlapping or anything:
Screen Shot 2024-05-09 at 12 03 33 PM

**Comment by taipeicoder** [comment]
@arthur791004 in mobile resolution, clicking the more icon in the row doesn't open the option menu, instead I get redirected to the domain manage page. Are you able to reproduce?

**Comment by arthur791004** [comment]
> in mobile resolution, clicking the more icon in the row doesn't open the option menu, instead I get redirected to the domain manage page. Are you able to reproduce?

Ooops...it's broken when I changed to use the domain-link overlay. Addressed it by https://github.com/Automattic/wp-calypso/pull/90441/commits/604aa1cbe359541fd3e2c3f0e652f24a2ae85805

**Review by taipeicoder — APPROVED** [review]
LGTM!

**Comment by arthur791004** [comment]
@roo2 I've pushed some fixes. Could you help to review it again when you have time? Thank you!

**Comment by Addison-Stavlo** [comment]
My concern is more on the design side and not so much our implementation. cc @lucasmendes-design 

But why should we make the entire row clickable here for an action that navigates you out of the page, when we are in an interface for selecting bulk actions. Imagine user flow:
* User is on /domains/manage to perform bulk actions.
* User is selecting sites with the checkbox to apply bulk management to.
* after selecting a handful of sites and going to click another they are 2px away from the checkbox (or imagine trying to 'click' mobile with your finger and having the click point be just outside the checkbox)
* User is then navigated to a separate page and lost all their selection work.

bad-click

This seems like an incredibly poor user experience, especially if we are considering mobile clicks for bulk management selections.

**Comment by lucasmendes-design** [comment]
> My concern is more on the design side and not so much on the implementation (...)

Hi there, sorry for the delay. I have two possible solutions for this issue. If it's not possible. I'd vote to make the whole row clickable so it's consistent with the Sites page. 

Approach 01: 
Can we make the clickable area around the checkbox bigger? Gmail also has something similar. 

image
image

Approach 02:
Can we make only the checkbox column not clickable? 
image

**Comment by davemart-in** [comment]
Adding `event.stopPropagation()` to the checkboxes should do the trick I believe.

File: packages/domains-table/src/domains-table/domains-table-row.tsx [hunk 2]
```diff
@@ -80,10 +85,21 @@ export function DomainsTableRow( { domain }: DomainsTableRowProps ) {
...
+		<tr
+			key={ domain.domain }
+			className="domains-table__row"
+			onClick={ domainManagementLink ? handleSelect : undefined }
+		>
 			{ canSelectAnyDomains && (
-				<td className="domains-table-checkbox-td">
+				<td
+					className="domains-table-checkbox-td"
+					onClick={ ( e: MouseEvent ) => e.stopPropagation() }
```
Comment (roo2) [inline packages/domains-table/src/domains-table/domains-table-row.tsx]: Ahh it looks like you already are disabling propogation on the TD, but the TD is actually not wide enough to catch those clicks.I think we can fix this by adding a fixed width to it


Screenshot 2024-05-13 at 12 28 57 pm

**Review by roo2 — APPROVED** [review]
LGTM!

I agree that we need to fix that issue with clicks on the TD, but that is actually already fixed in code, it's just that the TD is too narrow, so technically it's the nighbouring TD being clicked XD.

There are a few breakpoints where the table is forced to scroll. We should probably fix this at the same time as we increase the width of that TD, I'll make a card for it.

https://github.com/Automattic/wp-calypso/assets/22446385/fe75e6b3-3150-4e74-9705-40ec3a41817c

**Comment by roo2** [comment]
Created new card to handle TD width and horizaontal scrolling 7114-gh-Automattic/dotcom-forge

**Comment by arthur791004** [comment]
Yes, the current issue is the checkbox column is a bit narrow 😞

**Comment by arthur791004** [comment]
I made the following changes to increase the area of the checkbox
* Add the min-width to the checkbox column to avoid it auto-shrinking
* Add the `margin-inline-start: -8px` to allow the `padding-inline-start` to increase the width
* On mobile, make the row clickable only when the bulk actions are disabled. So, the user might not misclick on the row when they want to perform the bulk actions

**Comment by arthur791004** [comment]
I'm going to merge this and we can track the follow-up issues on https://github.com/Automattic/dotcom-forge/issues/7114.

**Comment by lucasmendes-design** [comment]
LGTM.

**Comment by Addison-Stavlo** [comment]
> Yes, the current issue is the checkbox column is a bit narrow 😞

I think that is more of a side issue. The main issue is we are using the primary interaction mechanism of this page to trigger a secondary action (navigating to a single domains page) instead of triggering the primary action (toggling the checkbox). That is the conflicting pattern that is source of trouble, not the size of the checkbox or any event propagations.

> Hi there, sorry for the delay. I have two possible solutions for this issue. If it's not possible. I'd vote to make the whole row clickable so it's consistent with the Sites page.

@lucasmendes-design thoughts for the future here. But given this is a "bulk" management page, having the entire row clickable is fine but it should represent the **primary action**: selecting a site for bulk management (toggling a checkbox). Given the bulk management context, navigating to a single domain page is a **secondary action**. Clicking an entire row is the primary interaction, and now we have a primary interaction that triggers a secondary action instead of the primary action of this page.

Given that we should STRONGLY consider:
* Make clicking the row a convenient way to select the checbkox and toggle the domain for bulk management.
* And add a (or add more clarity to the) secondary action link to navigate you to the single domain page.

**Comment by roo2** [comment]
hmmmmm good point @Addison-Stavlo.

We're just talking about Mobile, when bulk selection is enabled right? On desktop, I think the primary action is openning the domain to manage, and managing bulk selection is the secondary action.

On mobile, if bulk selection has been enabled, then I think you've got a very good point that the primary thing they are doing is selecting rows and you wouldn't want them to be interupted by misclicking

https://github.com/Automattic/wp-calypso/assets/22446385/220bd810-e875-47c9-9ada-535c48a26b07

**Pull request merged**